cloud/cloud.go

```diff
@@ -796,15 +800,26 @@ SENTINEL:
 			return fmt.Errorf("cloud server start up script could not be made executable: %s", err)
 		}
 
-		// *** currently we have no timeout on this, probably want one...
+		// protect running the script with a timeout
+		limit := time.After(pcsTimeOut)
+		exiterr := make(chan error, 1)
 		var stderr string
-		_, stderr, err = s.RunCmd(pcsPath, false)
-		if err != nil {
-			err = fmt.Errorf("cloud server start up script failed: %s", err.Error())
-			if len(stderr) > 0 {
-				err = fmt.Errorf("%s\nSTDERR:\n%s", err.Error(), stderr)
+		go func() {
+			var runerr error
+			_, stderr, runerr = s.RunCmd(pcsPath, false)
+			exiterr <- runerr
+		}()
+		select {
+		case err = <-exiterr:
+			if err != nil {
+				err = fmt.Errorf("cloud server start up script failed: %s", err.Error())
+				if len(stderr) > 0 {
+					err = fmt.Errorf("%s\nSTDERR:\n%s", err.Error(), stderr)
+				}
+				return err
 			}
-			return err
+		case <-limit:
+			return fmt.Errorf("cloud server start up script failed to complete within %s", pcsTimeOut)
```

Comment:
mention the time unit?

Reply:
The time unit will be part of the error message.